qcodes/dataset/dependencies.py

```diff
@@ -0,0 +1,50 @@
+from typing import Dict, Any
+
+from qcodes.dataset.param_spec import ParamSpec
+
+
+class InterDependencies:
+    """
+    Object containing the ParamSpecs of a given run
+    """
+
+    def __init__(self, *paramspecs: ParamSpec) -> None:
+
+        for paramspec in paramspecs:
+            if not isinstance(paramspec, ParamSpec):
+                raise ValueError('Got invalid input. All paramspecs must be '
+                                 f'ParamSpecs, but {paramspec} is of type '
+                                 f'{type(paramspec)}.')
+
+        self.paramspecs = paramspecs
+
+    def __repr__(self) -> str:
+        output = self.__class__.__name__
+        tojoin = (str(paramspec) for paramspec in self.paramspecs)
+        output += f'({", ".join(tojoin)})'
```

Comment:
does this mean that there is no space between `output` and `f'({", ".join(tojoin)})'` parts?